docs/RFC/drop_schema.md

```diff
@@ -0,0 +1,55 @@
+## Introduction
+
+This document describes a design for dropping SKV schemas and truncating the associated data records. It is non-transactional, destructive, and meant to be an admin operation (e.g. it won't be exposed on the SKV client API).
+
+### Terminology
+
+For this document:
+- Schema: Only the schema description, not the data that refers to that schema.
+- Truncate: A destructive delete, with no possibility of rollback and no ability to read old MVCC versions.
+
+### Use Cases
+
+- Benchmarking and testing. For example, dropping TPC-C tables to more quickly re-run the benchmark.
+- User wants to re-architect an application. The user is responsible for migrating traffic and data to make the operation safe even though it is non-transactional.
+- Quickly regain capacity in the cluster. Since the drop and truncate is non-transactional, the capacity is regained immediately instead of after retention window expiration.
+
+### Rationale
+
+In general, users and the system should not think of schemas like tables. Schemas do not own their data like tables do. They are merely a description of the data formats that can be stored on servers. So K23SI operations do not operate over schemas; instead schema operations are CPO operations, like create collection. We should think of schemas as a performance optimization over self-describing records.
+
+## Design: truncateDataAndDropSchema
+
+truncateDataAndDropSchema truncates all data and all schema references for a given schema name (i.e. it deletes all versions of the schema).
+
+Operations:
+1. User request: truncateDataAndDropSchema(name)
+2. Client: the client library translates the name to the schema ID
+3. Client: Request sent to CPO
+4. CPO: drops schema locally (will no longer be returned by getSchema requests)
+5. CPO: sends truncateDataAndDropSchema to all partitions
+6. Partition: remove schema locally (no longer accepts requests for that schema)
```

Comment:
At partition, schema exists only in memory. So let's make it clear that remove is for in memory only.